scripts/release_candidate.py

```
@@ -3587,12 +3587,33 @@ def _validate_windows_setup_pe(path: Path) -> tuple[str, bool]:
+    pkcs7_records = 0
+    while cursor < certificate_end:
+        if certificate_end - cursor < 8:
+            raise CandidateError("Windows Setup contains malformed WIN_CERTIFICATE bytes")
+        record_size, revision, certificate_type = struct.unpack_from("<IHH", payload, cursor)
+        record_end = cursor + record_size
+        if record_size <= 8 or record_end > certificate_end:
+            raise CandidateError("Windows Setup contains malformed WIN_CERTIFICATE bytes")
+        if revision != 0x0200 or certificate_type != 0x0002:
+            raise CandidateError("Windows Setup contains an unsupported WIN_CERTIFICATE record")
+        if payload[cursor + 8] != 0x30:
+            raise CandidateError("Windows Setup WIN_CERTIFICATE does not contain DER CMS data")
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)